spec/instant.html

```diff
@@ -534,8 +534,17 @@ <h1>ParseTemporalInstant ( _isoString_ )</h1>
       </emu-alg>
     </emu-clause>
 
-    <emu-clause id="sec-temporal-compareepochnanoseconds" aoid="CompareEpochNanoseconds">
-      <h1>CompareEpochNanoseconds ( _epochNanosecondsOne_, _epochNanosecondsTwo_ )</h1>
+    <emu-clause id="sec-temporal-compareepochnanoseconds" type="abstract operation">
+      <h1>
+        CompareEpochNanoseconds (
+          _epochNanosecondsOne_: a BigInt,
+          _epochNanosecondsTwo_: a BigInt,
+        ): either -1, 0, or 1
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd></dd>
+      </dl>
       <emu-alg>
         1. If _epochNanosecondsOne_ &gt; _epochNanosecondsTwo_, return 1.
```

Comment:
why does this AO not `return _epochNanosecondsOne_ - _epochNanosecondsTwo_`? a comparator can be any negative or positive integer, or zero - it doesn't need to be specifically 1 or -1

Reply:
That predates my involvement in Temporal, but I'd guess it's because these are BigInts.

Reply:
those subtract just fine

Reply:
I mean, yes, but if you're implementing it in C++ where BigInt is an object instance, it's not clear-cut that (pseudocode, not a real BigInt API) `return BigInt::Subtract(one, two).toDouble().unwrap();` would be faster or more convenient than `if (BigInt::GreaterThan(one, two)) return 1.0; if (BigInt::LessThan(one, two)) return -1.0; return 0.0;` (Although WebKit uses `__int128` here, so that leans more obviously in one direction.)

This is an editorial PR so I don't want to debate this here, but if you suspect that implementors might care about the performance of this, go ahead and open a new issue to get their feedback; the bar for changes remains high at this point though.

Reply:
Totally agree it's not part of this PR, but it'd be an editorial change so i'm not sure speed or convenience matters here?

Reply:
I don't think it would be an editorial change? Unless what you're proposing is to change this operation not to specifically return -1/0/+1, and then add steps to [Temporal.Instant.compare](https://tc39.es/proposal-temporal/#sec-temporal.instant.compare) and [Temporal.ZonedDateTime.compare](https://tc39.es/proposal-temporal/#sec-temporal.zoneddatetime.compare) like
  1. Let _compareResult_ be CompareEpochNanoseconds(..., ...).
  2. If _compareResult_ < 0, return **-1**<sub>𝔽</sub>.
  3. If _compareResult_ > 0, return **1**<sub>𝔽</sub>.
  4. Return **+0**<sub>𝔽</sub>.

Reply:
altho also it'd be fine for the compare methods to return more numbers as well - comparators need not only return zero and ones